src/charts/axis.js

```diff
@@ -279,23 +293,46 @@ export default function Axis (_container) {
     return Math.ceil(longestLabelApproxWidth / (config.markPanelWidth / labels.length))
   }
 
+  function skipXLAbels () {
+    if (!config.xTickSkip) {
+      return
+    }
+    const labels = cache.xAxisRoot
+      .select(".axis.x")
+      .selectAll(".tick text")
+
+    let xTickSkip = config.xTickSkip
+    if (config.xTickSkip === "auto") {
+      const labelsText = labels.nodes().map(node => node.textContent)
+      xTickSkip = getNumberOfLabelsToSkip(labelsText)
+    }
+    labels.style("display", (d, i) => (i % xTickSkip) ? "none" : "block")
```

Comment:
to fix the ES Lint error [no-confusing-arrow](https://eslint.org/docs/rules/no-confusing-arrow) you should do:

```js
labels.style("display", (d, i) => ((i % xTickSkip) ? "none" : "block"))
```